src/Middleware.php

```diff
@@ -186,7 +186,24 @@ public static function retry(
             return new RetryMiddleware($decider, $delay, $handler, $stats);
         };
     }
-
+    /**
+    * Middleware wrapper function that adds an invocation id header to requests.
+    * This is a uniquely generated UUID to identify initial and subsequent retries as part of a complete request lifecycle.
+    *
+    * @return callable
+    */
+    public static function invocationID()
```

Comment:
This is a nitpick, but our naming structure would probably render this as `invocationId` instead of `invocationID`.